shared/src/test/scala/org/adridadou/openlaw/vm/OpenlawExecutionEngineSpec.scala

```diff
@@ -728,6 +728,75 @@ class OpenlawExecutionEngineSpec extends FlatSpec with Matchers {
     }
   }
 
+  it should "look at sub execution results for variable type" in {
+    val mainTemplate = compile("""[[c:Clause("clause")]]""")
+    val clauseTemplate = compile(
+      """
+        |<%
+        |# Arranged Variables by Header
+        |==MOU Terms==
+        |[[Terms: Collection<Text>]]
+        |
+        |==MOU Parties==
+        |[[Party Info: Structure(
+        |    Party Name: Text;
+        |    Party Email: Identity
+        |    )]]
+        |[[#Parties: Collection<Party Info>]]
+        |%>
+        |
+        |\centered **__MEMORANDUM OF UNDERSTANDING__**
+        |
+        |**NOW, THEREFORE,** the undersigned parties (the "***Parties***") see mutual benefit in entering into a partnership and formalizing their cooperation over the matters and under the terms sufficiently described in this Memorandum of Understanding ("***MOU***"):
+        |
+        |{{#for each Term: Terms =>
+        |
+        |    ^^[[Term]]
+        |}}
+        |
+        |The Parties shall keep confidential and shall not divulge to any other party, without the other's prior written consent, any non-public information concerning this MOU and the cooperation contemplated hereby, treating such confidential information with the same due care as their own proprietary information.
+        |
+        |The Parties shall jointly operate and exchange information with the intention of completing the transactions contemplated hereby and within determined timeframes and reasonable best efforts.
+        |
+        |After a preliminary review period, not to exceed two (2) months from the date of mutual execution below, the Parties may enter into a legally-binding agreement regarding these matters and shall make reasonable best efforts to conduct a meeting regarding the same following such MOU review period.
+        |
+        |This MOU is not a legally-binding agreement.
+        |
+        |***__SIGNATORIES__***
+        |
+        |{{#for each Party: Parties =>
+        |
+        |**[[Party.Party Name | Uppercase]]**
+        |
+        |*__[[Party.Party Email | Signature]]__*
+        |Authorized Representative
+        |
+        |}}
+        |""".stripMargin)
+
+    engine.execute(mainTemplate, TemplateParameters(), Map(TemplateSourceIdentifier(TemplateTitle("clause")) -> clauseTemplate,TemplateSourceIdentifier(TemplateTitle("clause")) -> clauseTemplate)) match {
+      case Right(result) =>
+        val variable = VariableDefinition(VariableName("Parties"), Some(VariableTypeDefinition("Collection",Some(VariableTypeDefinition("Parefhuzegfty Info",None)))))
+        getCollection(variable, result, "")
```

Comment:
What about adding an assert here to check if the returned type is indeed a collection value type with the expected values? If the returned collection is empty, should it break the test?

Reply:
The collection will be empty because the internal format value is an empty string.
The test here is more to check that the call actually works, there was an issue with this call previously